phpBB/phpbb/content_visibility.php

```diff
@@ -216,7 +222,13 @@ public function get_visibility_sql($mode, $forum_id, $table_alias = '')
 		}
 		else
 		{
-			$where_sql .= $table_alias . $mode . '_visibility = ' . ITEM_APPROVED;
+			$field_name = ($mode === 'topic') ? 'topic_poster' : 'poster_id';
+			$visibility_query = $table_alias . $mode . '_visibility = ';
+
+			$where_sql .= '(' . $visibility_query . ITEM_APPROVED . ')';
+			$where_sql .= ' OR (';
+			$where_sql .= '(' . $visibility_query . ITEM_UNAPPROVED . ' OR ' . $visibility_query . ITEM_REAPPROVE . ')';
```

Comment:
This need a switched off, its not good for large databases. There is no optimization possible.
```php
			$visibility_query = $table_alias . $mode . '_visibility = ';

			$where_sql .= $visibility_query . ITEM_APPROVED;
			if ($this->config['show_items_unapproved'])
			{
				$field_name = ($mode === 'topic') ? 'topic_poster' : 'poster_id';
				$where_sql .= ' OR (';
				$where_sql .= $db->sql_in_set($visibility_query, array(ITEM_UNAPPROVED, ITEM_REAPPROVE));
				$where_sql .= ' AND ' . $table_alias . $field_name . ' = ' . ((int) $this->user->data['user_id']) . ')';
			}

Reply:
A config switch will be in my next PR, need to decide whther to default on (my preference) or off.

Reply:
Would it help sql optimisation if the visibility of unapproved posts to posters had a time limit (only unapproved posts in the last X days)?

Reply:
Even if it does. I believe it would be quite bad for users to have the post disappear. They could think that it had been deleted....